### DISCUSSION TITLE
compress theme project to zip file

### DISCUSSION THREAD

**Discussion by Hsins** [tailpress/tailpress#122] [Ideas]
As we know that any wp-theme could be compressed to a `*.zip` file for uploading to WordPress site. Would tailpress support the compressing feature?

It would be best practice that finally `*.zip` shouldn't contain the following files or folders:

- `node_modules`
- `resources`
- `package.json`, `package-lock.json`, `postcss.config.js`, `tailwind.config.js`

There're two approaches for reference:

1. remove or exclude those files when compressing.
2. change current directory structure as the structure of [gregsullivan/_tw](https://github.com/gregsullivan/_tw)

First approach is compatible for the current project but the second approach is more clean and engineering because it follow the principle of SoC (Separation of concerns).

**Comment by jeffreyvr** [comment]
You can also create a zip through git. Example:

```bash
git archive --format zip --output your-theme.zip main
```

Where `main` is the branch name. This should already exclude folders/files that are ignored by git, such as node_modules. Then the other files could be added to the `.gitattributes` file - and then they will also be excluded from the zip.

**Reply by ethan-tqa** [reply]
It's probably good to add this to README

**Comment by justingolden21** [comment]
I was using something like this (work in progress)

```
@ECHO OFF
tar.exe --exclude=node_modules -a -cf my-theme.zip my-theme
PAUSE
```

**Reply by justingolden21** [reply]
Tar comes with Windows 10+
